src/components/svgSetting/svgSetting.js

```diff
@@ -1,34 +1,86 @@
-import React, { Component } from 'react';
-import '../checkBoxSelection/checkBoxSelection.scss';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import "../checkBoxSelection/checkBoxSelection.scss";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { Svgo } from "../svgSetting/svgoConfig";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class SvgSetting extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: this.props.isSelected
-        };
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected,
+      arr: [],
+      svgoObject: { ...Svgo },
+      IndexOf: -1
+    };
+  }
+
+  handleSettingsClick = () => {
+    this.setState({
+      isClicked: !this.state.isClicked
+    });
+
+    this.setState({
+      IndexOf: this.props.svgOptions.findIndex(this.props.value)
+    });
```

Comment:
what is the rational for storing this in local state, is it being used anywhere else in this class. would a normal variable not be more suited?

Reply:
oh my bad I just followed the pattern I found on this function,

Reply:
its cool, better to critique your own code before you push